File: code-challenges/arrayReverse/javascript/arrayReverse.js
'use strict';

/**
 * Reverses elements in a given array without using the built in array.reverse() method.
 * @param { Array } arr - An array of items to be reversed.
 * @returns - A new array with the elements in reverse order. 
 */
let reverseArray = (arr) => {
  let output = [];

  for (let i = (arr.length - 1); i >= 0; i--) {

    output.push(arr[i]);

  };

  return output;
}

module.exports = reverseArray;

File: code-challenges/arrayReverse/python/joseph-zabaleta/array_reverse.py
def array_reverse(arr):
    """
    Reverses elements in a given list without using built in list.reverse() method

    Parameters:
    arr (List): An array of items to be reversed.

    Returns:
    List: A new list with the elements in reverse order

    """
    output = []

    idx = len(arr) - 1

    while idx >= 0:
        output.append(arr[idx])
        idx -= 1

    return output
